src/main/java/com/google/devtools/build/lib/rules/config/ConfigGlobalLibrary.java

```diff
@@ -97,8 +106,11 @@ private void validateBuildSettingKeys(
     for (String optionKey : optionKeys) {
       if (!optionKey.startsWith(COMMAND_LINE_OPTION_PREFIX)) {
         try {
-          var unused = Label.parseCanonicalUnchecked(optionKey);
-        } catch (IllegalArgumentException e) {
+          Label label = Label.parseWithRepoContext(optionKey, packageContext);
+          if (!label.getRepository().isVisible()) {
+            throw Starlark.errorf("invalid transition %s '%s'", singularErrorDescriptor, label);
```

Comment:
maybe add the whole clause about "repo X not visible from X"?

Reply:
Done